dozer/cogs/roles.py

```diff
@@ -5,11 +5,19 @@
 
 from ._utils import *
 from .. import db
+from .. import bot
 
 
 class Roles(Cog):
     """Commands for role management."""
 
+    def __init__(self, aa):
+        for command in self.giveme.walk_commands():
+            @command.before_invoke
+            async def givemeautopurge(self, ctx):
+                if await self.ctxpurge(ctx, False):
+                    await ctx.send("That role does not exist any more, removing from list")
```

Comment:
"That role" wouldn't make much sense for `&giveme list` or other commands that don't specify a role; and if, for example, a user ran `&giveme a, b` and the bot deleted a missing record that the user didn't request, seeing both "That role doesn't exist" and "Gave you two roles" indicates it both did and didn't complete successfully. "Purged {n} missing role(s)" would make more sense to send.